lib/internal/validators.js

```diff
@@ -180,6 +181,19 @@ function validateEncoding(data, encoding) {
   }
 }
 
+// Check that the port number is not NaN when coerced to a number,
+// is an integer and that it falls within the legal range of port numbers.
+function validatePort(port, name = 'Port', { allowZero = true } = {}) {
```

Comment:
Using the destructuring with default values is an overhead that each function call has to go through. I am not a fan of having magical boolean values but that would have less overhead.

Reply:
While there is certainly an overhead, it's been getting consistently better with V8 updates and it's consistent with style already used in the validators.js file (see https://github.com/nodejs/node/blob/a100a5460adf7db8939fffc2d81f82858fb6b6ad/lib/internal/validators.js#L133 for instance).

Reply:
Ping @BridgeAR ... given that these are consistent with the existing style in the validators.js, are you ok with moving forward?

Reply:
Ping @BridgeAR ... I'd like to get this moving forward as it's part of the prep work for updating the QUIC PR

Reply:
I personally would change it to the boolean as mentioned above and potentially also change the other occurrences in `validators.js`. I won't block the PR as it's likely not going to be a huge performance difference. As such it's fine to land this if others think it's fine.